src/python/resize/core.py

```diff
@@ -0,0 +1,33 @@
+from copy import copy
+from pathlib import Path
+from typing import Optional, Tuple
+
+import cv2
+from base import Image, read_image
+
+
+class IncorrectParameters(Exception):
+    def __init__(self, message):
+        self.message = message
+        super().__init__(self.message)
+
+
+def resize(
+    path: Path,
+    percentage: Optional[float] = None,
+    width_height: Optional[Tuple[int, int]] = None,
+) -> Image:
+    """Resizes a given image, `percentage` parameter preserves aspect ratio,
+    `width_height` does not.
+    """
+    if percentage and width_height:
+        raise IncorrectParameters("Don't provide both at the same time!")
+
+    img = read_image(path)
+
+    if percentage:
+        dimentions = (int(img.width * percentage), int(img.height * percentage))
+    else:
+        dimentions = copy(width_height)
+
+    return Image(file=cv2.resize(img.file, dimentions, interpolation=cv2.INTER_AREA))
```

Comment:
I would rather omit the custom error type and create a function called `resize`
```suggestion
def resize(
    im: Image,
    size: Tuple[int, int],
    interpolation=cv2.INTER_AREA,
    **kwargs,
) -> Image:
    """Resizes a given image to dimensions specified by `size=(width, height)`.
    Convenience wrapper around `cv2.resize`.
    """
    return Image(file=cv2.resize(im.file, size, interpolation=interpolation, **kwargs))


def scale(im: Image, scale: float = 1.0, **kwargs):
    """Rescales an image by some fixed proportion for both width and height.
    """
    size = (int(im.width * scale), int(im.height * scale))
    return resize(im, size, **kwargs)
```